### DISCUSSION TITLE
Set a default filter?

### DISCUSSION THREAD

**Discussion by tonypartridge** [rappasoft/laravel-livewire-tables#948] [Q&A] [answered]
Sorry if it has been asked, I need to set a dynamic default filter? For example, I want to filter all records to show this month when a month has not been selected manually. I can see how to do this with a public property, but they only work with static values. I'd like to use date and strtotime to set the value for this month.

**Comment by tonypartridge** [comment, marked as answer]
I've solved it with the below

```php
    public function mount() {
        if(!$this->getAppliedFilterWithValue('pay_period')) {
            $this->setFilter('pay_period', '2022-10-01');
        }
    }
```

**Reply by LWlook** [reply]
Hey @tonypartridge ,

with the setFilter-Methode you causing a little refresh and also don't really set the default filter.

Is there a way, if I click on "reset filter" that a chosen default is triggered?

I hope you understand what I mean :smile:

**Reply by LWlook** [reply]
And another thing: If I use your method, and go to another page (pagination) and refresh the page, than the page could be empty.
